src/darwin/Framework/CHIP/MTRDeviceControllerDelegate.h

```diff
@@ -54,6 +54,18 @@ API_AVAILABLE(ios(16.4), macos(13.3), watchos(9.4), tvos(16.4))
  */
 - (void)controller:(MTRDeviceController *)controller commissioningComplete:(NSError * _Nullable)error;
 
+/**
+ * Notify the delegate when commissioning is completed.
+ * For getting commissioning controllee's deviceId
```

Comment:
That's already known, because there is only one commissioning happening at a time, and the device id is the input, no?

We need to generally redesign how the commissioning notifications happen and have it not be a singleton...

Reply:
I know it's the same as the NodeID that was initially assigned, but if you assign another NodeID within the Commissionee's NOC, it can be a different NodeID (for example, when using setNocChainIssueer). I think it would be easier to add NodeID to design considering these various cases. (You can also choose because there are existing ones.)

Reply:
If the module that manages NOC generation and NodeID in setNocChainIssueer is configured separately, a value different from the original NodeID can be assigned to prevent duplication. The Controller will use the NodeID in the NOC. In this case, it is more appropriate to add the NodeID to CommissioningComplete because there is a possibility of assigning a different NodeID.

Reply:
> If the module that manages NOC generation and NodeID in setNocChainIssueer is configured separately

What does "configured separately" mean?  You mean that the operational credentials delegate and the device controller delegate are not talking to each other?

Please don't resolve this conversation; I'll resolve it once we come to a conclusion.

Reply:
To explain what I know,
1. The initial NodeID is set in the same way as the example in the code below.
https://github.com/project-chip/connectedhomeip/blob/7ccd5d40e4540e2a2ef1b24686a8128fe6c6d2ff/src/darwin/CHIPTool/CHIPTool/View%20Controllers/QRCode/QRCodeViewController.m#L500

2. If setNocChainIssuer is not set, NOC is created with assigned assignId value as shown in the code below.
https://github.com/project-chip/connectedhomeip/blob/7ccd5d40e4540e2a2ef1b24686a8128fe6c6d2ff/src/darwin/Framework/CHIP/MTROperationalCredentialsDelegate.mm#L287

However, if setNocChainIssuer is set, commissioning is performed with the NodeID in the NOC created in the Application Layer.

In my opinion, separating the NOC generation part and the part that manages the NodeID that has completed Commissioning makes it easier to implement the code by passing the NodeID at the time of CommissioningComplete.

My opinion is that separating the code of the NOC generation part and the Commissioned NodeID management part can implement a code that is easier.

Separate code implementation is possible by adding NodeID during CommissioningComplete.

Reply:
OK, right, so you have a situation where the operational credentials delegate (please don't use `setNocChainIssuer`; it's deprecated) and the controller delegate are not talking to each other.  Yeah, in that case adding the node ID here makes sense....  It's just unfortunate that we're adding more to the controller delegate API, which is broken in its current form and needs to be replaced anyway.